aya/src/maps/mod.rs

```diff
@@ -604,28 +622,32 @@ impl MapData {
     /// Returns the file descriptor of the map.
     ///
     /// Can be converted to [`RawFd`] using [`AsRawFd`].
-    pub fn fd(&self) -> Option<MapFd> {
-        self.fd.map(MapFd)
+    pub fn fd(&self) -> Option<MapFd<'_>> {
+        self.fd.as_ref().map(|fd| MapFd(fd.as_fd()))
     }
 }
 
-impl Drop for MapData {
-    fn drop(&mut self) {
-        // TODO: Replace this with an OwnedFd once that is stabilized.
-        if let Some(fd) = self.fd.take() {
-            unsafe { libc::close(fd) };
-        }
+impl MapData {
+    /// Attempts to duplicate MapData
+    ///
+    /// Fails if if the inner file descriptor could not be cloned
+    // TODO (AM): should we return MapError? New variant? Maybe make MapError non_exhaustive?
```

Comment:
The current enum is exhaustive so adding any variant would be a breaking change, I think we probably should make it non exhaustive and add a new variant, unless there is an existing one that people think already fits this? nothing jumped out to me.